client/payment-gateways/payment-gateways-confirmation.js

```diff
@@ -0,0 +1,105 @@
+/* global jQuery, woocommerce_admin */
+/**
+ * External dependencies
+ */
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import DisableConfirmationModal from './disable-confirmation-modal';
+import { useSettings } from 'data';
+
+const PaymentGatewaysConfirmation = () => {
+	// pre-fetching the settings (and available payment methods) _before_ the modal is displayed,
+	// so that there isn't a delay in the modal rendering
+	useSettings();
+
+	const [
+		isConfirmationModalVisible,
+		setIsConfirmationModalVisible,
+	] = useState( false );
```

Comment:
Same question about mixing local-state and app-wide state.

Reply:
I think in this case it is best to keep things into local state.
The modal is displayed within the component, no other component needs to know about the modal being opened/closed. And in this case the state is also not duplicated between local state/redux.
So I think colocating the state as close as possible as where it's used in this case is best, IMO :) 
Since the state is colocated, there's also no issue creating additional action types/action creators/action handler in the reducer, which also reduces possible issues with components consuming the redux state re-rendering or re-running hooks unnecessarily :)

Reply:
Ok, I get your point and it makes sense to me. Agree with you, in this case, is probably not worth it and doesn't make sense. :+1: